examples/eks/README.md

```diff
@@ -0,0 +1,226 @@
+e is 65537 (0x010001)
+Signature ok
+subject=CN = connaisseur-svc.connaisseur.svc
+Getting Private key
+kubectl create ns connaisseur || true
+namespace/connaisseur created
+kubectl config set-context --current --namespace connaisseur
+Context "your-username@your-clustername.your-region.eksctl.io" modified.
+helm install connaisseur helm --wait
+NAME: connaisseur
+LAST DEPLOYED: Wed Oct  7 10:45:53 2020
+NAMESPACE: connaisseur
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+```
+
+Connaisseur was successfully deployed.
+
+### (Optional) 5. Verify Connaisseur is working
+
+If you were just trusting everything someone told you, you wouldn't be here looking for a tool that ensures image integrity, so don't take our word for Connaisseur working. Go ahead and try it:
+
+`kubectl get all -n connaisseur` will show you most* of what you deployed (ConfigMaps and Secrets are not shown).
+
+Try deploying the images you created in step 2. First try deploying the unsigned image:
+
+```bash
+kubectl create -f - << EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-deployment
+  labels:
+    app: sample
+spec:
+  selector:
+    matchLabels:
+      app: sample
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sample
+    spec:
+      containers:
+      - name: sample
+        image: ${IMAGE_PATH}:unsigned
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: sample
+  labels:
+    app: sample
+spec:
+  ports:
+  - name: port
+    port: 8080
+    targetPort: 5000
+  selector:
+    app: sample
+  type: LoadBalancer
+EOF
+```
+
+You should see the deployment being rejected with an error similar to:
+
+```bash
+service/sample created
+Error from server: error when creating "STDIN": admission webhook "connaisseur-svc.connaisseur.svc" denied the request: could not find signed digest for image "docker.io/testingconny/testserver:unsigned" in trust data.
+```
+
+> Note that the service was indeed created, because it does not reference an image and is thus not blocked by Connaisseur. You can clean it up by calling `kubectl delete service sample`.
+
+Finally, make sure that Connaisseur will deploy the signed image and isn't just rejecting all images:
+
+```bash
+kubectl create -f - << EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-deployment
+  labels:
+    app: sample
+spec:
+  selector:
+    matchLabels:
+      app: sample
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sample
+    spec:
+      containers:
+      - name: sample
+        image: ${IMAGE_PATH}:signed
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: sample
+  labels:
+    app: sample
+spec:
+  ports:
+  - name: port
+    port: 8080
+    targetPort: 5000
+  selector:
+    app: sample
+  type: LoadBalancer
+EOF
+```
```

Comment:
this and the following could technically also be example files, right?

Reply:
technically yes, but then you do need envsubst to substitute the IMAGE_PATH variable and envsubst is actually not installed by default and iirc needs quite a few dependencies. All the other tooling, git, make, kubectl, sed, grep, ... are either preinstalled on most systems or can be reasonably suspected to be on the machine of whoever tries the tutorial. Envsubst isnt, so I wouldnt do it

Reply:
Also tbh I like the fact that we show what's being run on k8s, instead of `kubecl apply -f thisfilewhichmayrunarbitraryprivilegeddeployments.yaml`. For the Dockerfiles it's not that important, since it will by default be run in the very deployment defined in this piece of code and thus can be considered somewhat sandboxed